src/AudioManager.ts

```diff
@@ -0,0 +1,147 @@
+  streamTimeslice?: number;
+  handleStartSTT: () => void;
+  handleStopSTT: () => void;
+  handleTranscript: (transcript: string) => void;
+  handleError?: (error: string) => void;
+}
+
+class AudioManager {
+  private audioInputsService: AudioInputsService;
+
+  private audioInputsBrowser: AudioInputsBrowser;
+
+  private audioOutputsService: AudioOutputsService;
+
+  private mediaAudio: MediaAudio;
+
+  private duckVolumeLevel: number;
+
+  private normalVolumeLevel: number;
+
+  private sttService: "browser" | "charisma/deepgram";
+
+  constructor(options: AudioManagerOptions) {
+    this.duckVolumeLevel = options.duckVolumeLevel ?? 0;
+    this.normalVolumeLevel = options.normalVolumeLevel ?? 1;
+    this.sttService = options.sttService ?? "charisma/deepgram";
+
+    this.audioInputsService = new AudioInputsService(options.streamTimeslice);
+    this.audioInputsBrowser = new AudioInputsBrowser();
+    this.audioOutputsService = new AudioOutputsService();
+    this.mediaAudio = new MediaAudio();
+
+    // Listen to events from the AudioInputsService
+    this.audioInputsService.on("start", options.handleStartSTT);
+    this.audioInputsService.on("stop", options.handleStopSTT);
+    this.audioInputsService.on("transcript", options.handleTranscript);
+    this.audioInputsService.on("error", options.handleError ?? console.error);
+
+    // Listen to events from the AudioInputsBrowser
+    this.audioInputsBrowser.on("start", options.handleStartSTT);
+    this.audioInputsBrowser.on("stop", options.handleStopSTT);
+    this.audioInputsBrowser.on("transcript", options.handleTranscript);
+    this.audioInputsBrowser.on("error", options.handleError ?? console.error);
+  }
+
+  // **
+  // ** Audio Input ** //
+  // **
+  public startListening = (): void => {
+    if (this.sttService === "browser") {
+      this.audioInputsBrowser.startListening();
+    } else if (this.sttService === "charisma/deepgram") {
+      this.audioInputsService.startListening();
+    }
+
+    if (this.mediaAudio.isPlaying) {
+      this.mediaAudio.volume = this.duckVolumeLevel;
+    }
+  };
+
+  public stopListening = (): void => {
+    if (this.sttService === "browser") {
+      this.audioInputsBrowser.stopListening();
+    } else if (this.sttService === "charisma/deepgram") {
+      this.audioInputsService.stopListening();
+    }
+
+    if (this.mediaAudio.isPlaying) {
+      this.mediaAudio.volume = this.normalVolumeLevel;
+    }
+  };
+
+  public connect = (token: string): void => {
+    if (this.sttService === "charisma/deepgram") {
+      this.audioInputsService.connect(token);
+    }
+  };
+
+  public resetTimeout = (timeout: number): void => {
+    if (this.sttService === "charisma/deepgram") {
+      this.audioInputsService.resetTimeout(timeout);
+    } else {
+      this.audioInputsBrowser.resetTimeout(timeout);
+    }
+  };
+
+  // **
+  // ** Browser STT Service ** //
+  // **
+  public browserIsSupported = (): boolean => {
```

Comment:
what should this do intuitively if browserIsSupported is called when I'm only (intending to) use audioInputsService ?

Reply:
Either expect that I've read the docs and know that I don't need this function for still work as expected either way. i.e return true if the browser is supported. The user will have set that sttService they are using already